client/components/search/index.jsx

```diff
@@ -38,7 +38,8 @@ module.exports = React.createClass( {
 		onKeyDown: React.PropTypes.func,
 		disableAutocorrect: React.PropTypes.bool,
 		onBlur: React.PropTypes.func,
-		searching: React.PropTypes.bool
+		searching: React.PropTypes.bool,
+		forceCloseButton: React.PropTypes.bool
```

Comment:
Adding in this case does perform a search. It first finds the site you typed in and shows you a preview prior to committing to that. It seems like a pretty valid reuse of the search component.

I am only confused as to why the add interface doesn't think of itself as 'pinned' just like the search. You click a button to open it just as you do search in the `SectionNav`. That said, I'm late to this conversation and can be ignored.